extensions/amp-analytics/0.1/instrumentation.js

```diff
@@ -212,14 +213,21 @@ export class InstrumentationService {
    * @private
    */
   createVisibilityListener_(callback, config) {
-    if (config['visibilitySpec']) {
+    const spec = config['visibilitySpec'];
+    const selector = config['selector'] || (spec && spec['selector']);
+    const analyticsVariables = config[vars] || {};
+    if (spec) {
       if (!isVisibilitySpecValid(config)) {
         return;
       }
+      const el = this.win_.document.getElementById(selector
```

Comment:
this and following three lines can be moved inside the callback passed into the listenOnce call below(#L231).